bin/crew

```diff
@@ -130,6 +130,8 @@ at_exit do
 
   # Print exit messages.
   ExitMessage.print
+  # Replace the currently running Ruby process with a new instance of the parent shell (thus sourcing the relevant configuration)
+  exec(`ps -p #{Process.ppid} -o comm=`.chomp) unless CREW_SCRIPTING
```

Comment:
Bump crew version in `lib/const.rb`?

Reply:
So this approach doesn't work, right? So do we need to just rename the print_source_bashrc line with something else since the autosource PR handles this? Maybe `'s/print_source_bashrc/source_bashrc/'` or  `'s/print_source_bashrc/autosource_bashrc/'` ?

Reply:
i'll see if i can profile that, actually. some numbers might help in the decision.

Reply:
looks like each new bash process adds around 8MB. That's not acceptable, I'll look into ways of dealing with that.

Reply:
Slightly more advanced testing setup (copilot-written script, so take it with a grain of salt)
```bash
#!/bin/bash

while true; do
    bash -c 'echo $$; ps -p $$ -o rss= | awk "{ printf \"%.2f MB\n\", \$1/1024 }"'
    sleep 3
done
```

Gives around 3.5 MB per new process.

Reply:
+1 for renaming `print_source_bashrc` to `source_bashrc`.  This will enable us to control at a package level which installation requires a `source ~/.bashrc`.  That said, this PR has a lot of changes to make that happen.

Reply:
If we can `source ~/.bashrc` on crew exit, which we can in master, why don't we just do that unconditionally?

Reply:
It doesn't make sense to source unless it would be necessary.

Reply:
Better to be safe than sorry, and its not like sourcing costs us anything. Maybe only run after installing a package?

Reply:
Sourcing after every install could have unintended consequences.  For instance, sometimes you might want to set environment variables for a specific use case and the sourcing could revert those settings.  Better to have control on a package level.

Reply:
By that logic, sourcing after any install could have unintended consequences-- better to consistently do something than have seemingly random failures of environment variables and the like.

Reply:
You're not understanding my point.  Having the boolean property lets you have control over the process in cases where you might not want to source.  It also makes packages that require sourcing self-documented so you know when it is needed.  How is that logic?

Reply:
How about checking if anything in `/usr/local/etc/env.d/` was changed, so that sourcing is only applied when it is needed?

Reply:
Is that not harder to do? And how do you know if ~/.bashrc has been sourced yet at all? Isn't that the larger problem?

Reply:
> Is that not harder to do? 
You check the latest file modification timein `/usr/local/etc/env.d/`, compare it with current, and done. Shouldn't be too bad. At any rate, I'm trying to avoid repeating lots of code in package files, and this is a good place to improve.

> And how do you know if ~/.bashrc has been sourced yet at all?
Not really sure what you mean here.

Reply:
>> And how do you know if ~/.bashrc has been sourced yet at all?
> Not really sure what you mean here.

Sometimes we want to make sure that the bashrc is sourced after the crew command, irrespective of whether files have changed in env.d (because scripts there might change some state when being run again, even if the scripts themselves haven't changed.)

Does that make sense?

Reply:
I think so? We can set it up to source after every command, but we currently don't. If we want to do it unconditionally, it shouldn't be too hard.

Reply:
I think there is a middle ground where we don't want the running to be unconditional. See for instance the bash scripts for distcc.

We don't need those run after every single crew command. They only need to be run after distcc is installed.

Reply:
(but they can be run after every shell invocation... That's fine... If run with every crew command we will essentially get logspam with every crew command.)